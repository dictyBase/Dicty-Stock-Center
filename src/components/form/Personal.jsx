// @flow
import React, { Component } from "react"
import FormGroupInput from "./FormGroupInput"
import { RequiredText } from "styles"

type Props = {
  firstName: Object,
  lastName: Object,
  email: Object
}

export default class Personal extends Component<Props> {
  render() {
    const { firstName, lastName, email } = this.props
    return (
      <div>
        <FormGroupInput field={firstName}>
          <RequiredText>* </RequiredText>
          First Name:
        </FormGroupInput>
        <FormGroupInput field={lastName}>
          <RequiredText>* </RequiredText>
          Last Name:
        </FormGroupInput>
        <FormGroupInput field={email}>
          <RequiredText>* </RequiredText>
          Email:
        </FormGroupInput>
      </div>
    )
  }
}
