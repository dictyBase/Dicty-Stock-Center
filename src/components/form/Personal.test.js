import React from "react"
import renderer from "react-test-renderer"
import "jest-styled-components"
import Personal from "./Personal"

test("matching a snapshot of Personal", () => {
  const firstName = {
    value: "Jane",
    touched: true
  }
  const lastName = {
    value: "Doe",
    touched: true
  }
  const email = {
    value: "janedoe@gmail.com",
    touched: true
  }

  const component = renderer.create(
    <Personal firstName={firstName} lastName={lastName} email={email} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
