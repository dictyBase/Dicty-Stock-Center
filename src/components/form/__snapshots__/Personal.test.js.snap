// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Personal 1`] = `
<div>
  <div
    className="sc-kLIISr itrVsn"
  >
    <div
      className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa iplmxF"
        is={null}
      >
        <label
          className="sc-dliRfk eLcvEu"
        >
          <span
            className="sc-hmXxxW fwNEuV"
          >
            * 
          </span>
          First Name:
        </label>
      </div>
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa cvzxNy"
        is={null}
      >
        <input
          className="sc-qrIAp hbjCtO"
          placeholder={undefined}
          type="text"
          value="Jane"
        />
      </div>
    </div>
  </div>
  <div
    className="sc-kLIISr itrVsn"
  >
    <div
      className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa iplmxF"
        is={null}
      >
        <label
          className="sc-dliRfk eLcvEu"
        >
          <span
            className="sc-hmXxxW fwNEuV"
          >
            * 
          </span>
          Last Name:
        </label>
      </div>
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa cvzxNy"
        is={null}
      >
        <input
          className="sc-qrIAp hbjCtO"
          placeholder={undefined}
          type="text"
          value="Doe"
        />
      </div>
    </div>
  </div>
  <div
    className="sc-kLIISr itrVsn"
  >
    <div
      className="sc-bxivhb bUZGRr sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
      is={null}
    >
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa iplmxF"
        is={null}
      >
        <label
          className="sc-dliRfk eLcvEu"
        >
          <span
            className="sc-hmXxxW fwNEuV"
          >
            * 
          </span>
          Email:
        </label>
      </div>
      <div
        className="sc-ifAKCX cFlEyZ sc-bdVaJa cvzxNy"
        is={null}
      >
        <input
          className="sc-qrIAp hbjCtO"
          placeholder={undefined}
          type="text"
          value="janedoe@gmail.com"
        />
      </div>
    </div>
  </div>
</div>
`;
